translator/translate/util/placeholderUtil.go

```diff
@@ -90,10 +93,22 @@ func GetMetadataInfo(provider MetadataInfoProvider) map[string]string {
 		accountID = unknownAccountId
 	}
 
-	return map[string]string{instanceIdPlaceholder: instanceID, hostnamePlaceholder: hostname,
-		localHostnamePlaceholder: localHostname, ipAddressPlaceholder: ipAddress, awsRegionPlaceholder: awsRegion,
-		accountIdPlaceholder: accountID,
+	metadata := map[string]string{
+		instanceIdPlaceholder:    instanceID,
+		hostnamePlaceholder:      hostname,
+		localHostnamePlaceholder: localHostname,
+		ipAddressPlaceholder:     ipAddress,
+		awsRegionPlaceholder:     awsRegion,
+		accountIdPlaceholder:     accountID,
 	}
+
+	tags := provider().Tags
+	for tagKey, tagValue := range tags {
+		tagPlaceholder := fmt.Sprintf("{tag:%s}", tagKey)
+		metadata[tagPlaceholder] = tagValue
```

Comment:
I think we're very fortunate that instance tags are not allowed to use `{` and `}` characters in the key name when pulling from IMDS, because I was going to say that this needed to be changed to guard against that, otherwise. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#tag-restrictions

Reply:
Thanks for this comment! I did consider the possible requirement to do some validation - however, as you're saying, we shouldn't need it in this case.